interpreter/runtimecost.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package interpreter provides functions to evaluate parsed expressions with
+// the option to augment the evaluation with inputs and functions supplied at
+// evaluation time.
+
+package interpreter
+
+import (
+	"github.com/google/cel-go/checker"
+	"github.com/google/cel-go/common/overloads"
+	"github.com/google/cel-go/common/types/ref"
+	"math"
+	"reflect"
+)
+
+// ActualCostEstimator is the interface to calculate cost for wired in function
+type ActualCostEstimator interface {
+	CallCost(overloadId string, args []ref.Val) *uint64
```

Comment:
+godoc, maybe "CallCost returns an estimated cost for the function overload invocation with the given args, or nil if it has not estimate to provide. CEL attempts to provide reasonable estimates for it's standard function library, so CallCost should typically not need to provide an estimate for CELs standard function." ?